[WebGPU] Implement GPURenderPassEncoder.setBlendConstant

mwyrzykowski opened the pull request:
#### c100cdc5c22b5437fea6ba4ec1384df12250f6d1
<pre>
[WebGPU] Implement GPURenderPassEncoder.setBlendConstant
<a href="https://bugs.webkit.org/show_bug.cgi?id=249802">https://bugs.webkit.org/show_bug.cgi?id=249802</a>
&lt;radar://103648024&gt;

Reviewed by Myles C. Maxfield.

Implement setBlendConstant which directly maps to
-[MTLRenderCommandEncoder setBlendColorRed:green:blue:alpha:]

* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::setBlendConstant):

Canonical link: <a href="https://commits.webkit.org/259005@main">https://commits.webkit.org/259005@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/224106fd4f8435411b4fed210bcf8a4e2b14b304

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103609 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12724 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36563 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112842 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13750 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3623 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95860 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/112008 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109381 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/10594 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/93660 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/38323 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92418 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25263 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79981 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6100 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26664 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6281 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/3188 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12262 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46175 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6185 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8033 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

djg reviewed (approved):
:shipit:

litherum commented on Source/WebGPU/WebGPU/RenderPassEncoder.mm:
```diff
@@ -181,7 +181,7 @@
 void RenderPassEncoder::setBlendConstant(const WGPUColor& color)
 {
-    UNUSED_PARAM(color);
+    [m_renderCommandEncoder setBlendColorRed:color.r green:color.g blue:color.b alpha:color.a];
```
No bounds checking?

Also, this should really have a real test, not a demo.

mwyrzykowski replied:
What type of bounds checking would you apply to colors? They can be any value positive or negative. We could make sure we don't pass NaNs or infs but I'm not sure if that is explicitly prohibited

mwyrzykowski replied:
Oh wait WGPUColor is double whereas these are floats

mwyrzykowski replied:
We could check for the values > FLT_MAX, < -FLT_MAX but would we abandon the operation or pass +/-inf? If we pass the inf values then there is no need to check

mwyrzykowski committed:
[WebGPU] Implement GPURenderPassEncoder.setBlendConstant
https://bugs.webkit.org/show_bug.cgi?id=249802
<radar://103648024>

Reviewed by Myles C. Maxfield.

Implement setBlendConstant which directly maps to
-[MTLRenderCommandEncoder setBlendColorRed:green:blue:alpha:]

* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::setBlendConstant):

Canonical link: https://commits.webkit.org/259005@main
